pkg/provider/loadBalancer.go

```diff
@@ -120,7 +112,98 @@ func syncLoadBalancer(ctx context.Context, kubeClient kubernetes.Interface, serv
 		}
 		return &service.Status.LoadBalancer, nil
 	}
+	return nil, nil
+}
+
+func parseAddrList(inputString string) (addrs []netip.Addr, err error) {
+	addrStringList := strings.Split(inputString, ",")
+	var addrList []netip.Addr
+
+	for i := range addrStringList {
+		addrString := addrStringList[i]
+		addr, err := netip.ParseAddr(addrString)
+		if err != nil {
+			return nil, err
+		}
+		addrList = append(addrList, addr)
+	}
+
+	return addrList, nil
+}
+
+// Gather infos about implemented services
+func mapImplementedServices(svcs *v1.ServiceList, mapSharedIpv4 bool) (inUseSet *netipx.IPSet, servicePortMap map[string]*set.Set[int32], err error) {
```

Comment:
```suggestion
func mapImplementedServices(svcs *v1.ServiceList, allowShare bool) (inUseSet *netipx.IPSet, servicePortMap map[string]*set.Set[int32], err error) {
```
Since this is just generic field to check whether shares are allowed, you have logic to check whether it's ipv4 address or not.